src/thx/Position.hx

```diff
@@ -0,0 +1,86 @@
+package thx;
+
+import haxe.PosInfos;
+
+abstract Position(PosInfos) from PosInfos to PosInfos{
+  public function new(self:PosInfos){
+    this = self;
+  }
+  @:noUsing static public function here(?pos:PosInfos):Position{
+    return new Position(pos);
+  } 
+  @:to public function toString():String{
+    return Positions.toString(this);
+  }
+  public var fileName(get,never)     : String;
+  public function get_fileName(){
+    return this.fileName;
+  }   
+  public var className(get,never)    : String;
+  public function get_className(){
+    return this.className;
+  }     
+  public var methodName(get,never)   : String;
+  public function get_methodName(){
+    return this.methodName;
+  }      
+  public var lineNumber(get,never)   : Int;
+  public function get_lineNumber(){
+    return this.lineNumber;
+  }     
+  public var customParams(get,never) : CustomParams;
+  public function get_customParams(){
+    return this.customParams == null ? this.customParams = new CustomParams(this.customParams) : this.customParams;
+  }
+
+  public function clone():Position{
+    return Positions.clone(this);
+  }
+  public function getClassMethodString():String{
+    return Positions.getClassMethodString(this);
+  }
+  public function getClassMethodLineString():String{
+    return Positions.getClassMethodLineString(this);
+  }
+}
+@:forward abstract CustomParams(Null<Array<Dynamic>>) from Null<Array<Dynamic>> to Null<Array<Dynamic>>{
```

Comment:
If we decide to keep this type why don't we try to properly type the custom params instead of using `Dynamic`?